src/details/http_endpoint.cpp

```diff
@@ -142,6 +142,7 @@ http_endpoint& http_endpoint::operator =(const http_endpoint& h) {
 }
 
 bool http_endpoint::operator <(const http_endpoint& b) const {
+    if (family_url != b.family_url) return family_url;
```

Comment:
Makes sense. I think we'd benefit from a few tests showing the behavior of family covering for already registered resourced while also showing that the specific resources are still called properly.

Reply:
Figuring out how to do that is why this PR is still a draft. I think I've also found another spot where I need to tweak things a bit to make this correct. (In`webserver::register_resource` don't insert into `registered_resources_str` if `family`, but again, I still don't have tests for that.)

One things I'm running into re: tests is that all the integration tests try to run on the same hard coded port and I keep getting `Unable to connect daemon to port: 8080`. I'm real tempted to try to make that auto select a port but I'm not sure how much I want to dig around to figure out how to get said port back out.